tests/api/tests_auth.py

```diff
@@ -0,0 +1,20 @@
+import os
+
+
+def test_public_endpoint(client):
+    response = client.get("/public")
+
+    assert response.status_code == 200
+
+
+def test_private_endpoint_without_api_key(client):
+    response = client.get("/private")
+    # for now we are not using api key
+    assert response.status_code == 200
+
+
+def test_private_endpoint_with_valid_api_key(client):
+    os.environ["API_KEY"] = "secret"
+    headers = {"x-api-key": "secret"}
+    response = client.get("/private", headers=headers)
+    assert response.status_code == 200
```

Comment:
```suggestion
@lru_cache()
def settings_api_key():
    return Settings(api_key="secret")


def test_private_endpoint_with_valid_api_key(fastapp, client):
    fastapp.dependency_overrides[get_settings] = settings_api_key

    headers = {"x-api-key": "secret"}
    with pytest.raises(HttpError403): # to be fixed wit accurate exception name
        client.get("/private")
    response = client.get("/private", headers=headers)
    assert response.status_code == 200
```

If it doesn't work, review this file for inspiration: https://github.com/scality/gh-actions-exporter/blob/main/tests/api/metrics/conftest.py